### ISSUE TITLE
sub: libuavcan dsdl failed to generate with StopIterator

### ISSUE THREAD

**Issue by patrickelectric** [ArduPilot/ardupilot#10333]
### Bug report

**Issue details**

Can't build with waf to Pixhawk1.

Libuavcan failed with:
```
libuavcan_dsdl_compiler.DsdlCompilerException: generator raised StopIteration
generator raised StopIteration
```

I'm running python 3.7.2;

**Version**
master - ff52ceabba4505999a5795d56606893eb1e00150

**Platform**
[  ] All
[  ] AntennaTracker
[  ] Copter
[  ] Plane
[  ] Rover
[x] Submarine

**Hardware type**
Pixhawk1 (Moved from px4-v2)

**Logs**
```sh
 patrick@pichau  /tmp  git clone github:ardupilot/ardupilot
Cloning into 'ardupilot'...
remote: Enumerating objects: 26, done.
remote: Counting objects: 100% (26/26), done.
remote: Compressing objects: 100% (20/20), done.
remote: Total 243851 (delta 9), reused 18 (delta 6), pack-reused 243825
Receiving objects: 100% (243851/243851), 121.22 MiB | 7.93 MiB/s, done.
Resolving deltas: 100% (179057/179057), done.
 patrick@pichau  /tmp  cd ardupilot 
 patrick@pichau  /tmp/ardupilot   master  git submodule update --init --recursive
Submodule 'modules/ChibiOS' (git://github.com/ArduPilot/ChibiOS.git) registered for path 'modules/ChibiOS'
Submodule 'modules/gbenchmark' (git://github.com/google/benchmark.git) registered for path 'modules/gbenchmark'
Submodule 'gtest' (git://github.com/ArduPilot/googletest) registered for path 'modules/gtest'
Submodule 'modules/mavlink' (git://github.com/ArduPilot/mavlink) registered for path 'modules/mavlink'
Submodule 'modules/uavcan' (git://github.com/ArduPilot/uavcan.git) registered for path 'modules/uavcan'
Submodule 'modules/waf' (git://github.com/ArduPilot/waf.git) registered for path 'modules/waf'
Cloning into '/tmp/ardupilot/modules/ChibiOS'...
Cloning into '/tmp/ardupilot/modules/gbenchmark'...
Cloning into '/tmp/ardupilot/modules/waf'...
Cloning into '/tmp/ardupilot/modules/gtest'...
Cloning into '/tmp/ardupilot/modules/uavcan'...
Cloning into '/tmp/ardupilot/modules/mavlink'...
Submodule path 'modules/ChibiOS': checked out '35ad8a79576e793114dd9288fc12c06d93a151d1'
Submodule path 'modules/gbenchmark': checked out '006d23ccca1375a973b7fae0cc351cedb41b812a'
Submodule path 'modules/gtest': checked out 'c99458533a9b4c743ed51537e25989ea55944908'
Submodule path 'modules/mavlink': checked out 'a9da2d2efa6a7d16004d5cde11ce1559aab645db'
Submodule 'pymavlink' (git://github.com/ArduPilot/pymavlink.git) registered for path 'modules/mavlink/pymavlink'
Cloning into '/tmp/ardupilot/modules/mavlink/pymavlink'...
Submodule path 'modules/mavlink/pymavlink': checked out 'c18db673f0214b683bc330587bc11ce2818a8e8d'
Submodule path 'modules/uavcan': checked out '72d4b9aecb5b23a142827bea962f636423d99041'
Submodule 'dsdl' (https://github.com/UAVCAN/dsdl) registered for path 'modules/uavcan/dsdl'
Submodule 'libuavcan/dsdl_compiler/pyuavcan' (https://github.com/UAVCAN/pyuavcan) registered for path 'modules/uavcan/libuavcan/dsdl_compiler/pyuavcan'
Cloning into '/tmp/ardupilot/modules/uavcan/dsdl'...
Cloning into '/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/pyuavcan'...
Submodule path 'modules/uavcan/dsdl': checked out '192295c4f9b67f4a20b0eabf74757b6597415f2b'
Submodule path 'modules/uavcan/libuavcan/dsdl_compiler/pyuavcan': checked out 'c58477a644d20ccf95a20c151f3a0402f271c3b8'
Submodule 'dsdl' (https://github.com/UAVCAN/dsdl) registered for path 'modules/uavcan/libuavcan/dsdl_compiler/pyuavcan/dsdl'
Cloning into '/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/pyuavcan/dsdl'...
Submodule path 'modules/uavcan/libuavcan/dsdl_compiler/pyuavcan/dsdl': checked out 'fd12483ddd4e58242d61d74a163e7aeaa1e0f466'
Submodule path 'modules/waf': checked out '337b0a96beeac517ecde081fb76dedd039089734'
patrick@pichau  /tmp/ardupilot   master  ./waf configure --board Pixhawk1
Setting top to                           : /tmp/ardupilot 
Setting out to                           : /tmp/ardupilot/build 
Autoconfiguration                        : enabled 
Setting board to                         : Pixhawk1 
Using toolchain                          : arm-none-eabi 
Checking for 'g++' (C++ compiler)        : /opt/gcc-arm-none-eabi-6-2017-q2-update/bin/arm-none-eabi-g++ 
Checking for 'gcc' (C compiler)          : /opt/gcc-arm-none-eabi-6-2017-q2-update/bin/arm-none-eabi-gcc 
Checking for c flags '-MMD'              : yes 
Checking for cxx flags '-MMD'            : yes 
Checking for program 'make'              : /usr/bin/make 
Checking for program 'arm-none-eabi-objcopy' : /opt/gcc-arm-none-eabi-6-2017-q2-update/bin/arm-none-eabi-objcopy 
Including /tmp/ardupilot/libraries/AP_HAL_ChibiOS/hwdef/fmuv3/hwdef.dat
Setup for MCU STM32F427xx
Writing hwdef setup in /tmp/ardupilot/build/Pixhawk1/hwdef.h
Writing DMA map
Generating ldscript.ld
Checking for env.py
env set HAL_WITH_UAVCAN=1
env set PROCESS_STACK=0x2000
env set MAIN_STACK=0x400
env set IOMCU_FW=0
env set CPU_FLAGS=['-mcpu=cortex-m4', '-mfpu=fpv4-sp-d16', '-mfloat-abi=hard', '-u_printf_float']
env set APJ_BOARD_ID=9
env set APJ_BOARD_TYPE=STM32F427xx
env set FLASH_RESERVE_START_KB=16
env set CHIBIOS_BUILD_FLAGS=USE_FATFS=yes CHIBIOS_STARTUP_MK=os/common/startup/ARMCMx/compilers/GCC/mk/startup_stm32f4xx.mk CHIBIOS_PLATFORM_MK=os/hal/ports/STM32/STM32F4xx/platform.mk MCU=cortex-m4 ENV_UDEFS=-DCHPRINTF_USE_FLOAT=1
Enabling ChibiOS asserts                     : no 
Checking for intelhex module:                : disabled 
Checking for HAVE_CMATH_ISFINITE             : yes 
Checking for HAVE_CMATH_ISINF                : yes 
Checking for HAVE_CMATH_ISNAN                : yes 
Checking for NEED_CMATH_ISFINITE_STD_NAMESPACE : yes 
Checking for NEED_CMATH_ISINF_STD_NAMESPACE    : yes 
Checking for NEED_CMATH_ISNAN_STD_NAMESPACE    : yes 
Checking for header endian.h                   : not found 
Checking for header byteswap.h                 : not found 
Checking for HAVE_MEMRCHR                      : no 
Checking for program 'python'                  : /usr/bin/python 
Checking for python version >= 2.7.0           : 3.7.2 
Checking for program 'python'                  : /usr/bin/python 
Checking for python version >= 2.7.0           : 3.7.2 
Source is git repository                       : yes 
Update submodules                              : yes 
Checking for program 'git'                     : /usr/bin/git 
Gtest                                          : STM32 boards currently don't support compiling gtest 
Checking for program 'arm-none-eabi-size'      : /opt/gcc-arm-none-eabi-6-2017-q2-update/bin/arm-none-eabi-size 
Benchmarks                                     : disabled 
Unit tests                                     : disabled 
Scripting                                      : disabled 
Scripting runtime checks                       : enabled 
Checking for program 'rsync'                   : not found 
'configure' finished successfully (0.774s)
 patrick@pichau  /tmp/ardupilot   master  ./waf build sub
Waf: Entering directory `/tmp/ardupilot/build/Pixhawk1'
b'Embedding file io_firmware.bin:Tools/IO_Firmware/fmuv2_IO.bin'
b'Embedding file bootloader.bin:/tmp/ardupilot/Tools/bootloaders/Pixhawk1_bl.bin'
Checking for env.py
env added HAL_WITH_UAVCAN=1
env added PROCESS_STACK=0x2000
env added MAIN_STACK=0x400
env added IOMCU_FW=0
env appended CPU_FLAGS=['-mcpu=cortex-m4', '-mfpu=fpv4-sp-d16', '-mfloat-abi=hard', '-u_printf_float']
env added APJ_BOARD_ID=9
env added APJ_BOARD_TYPE=STM32F427xx
env added FLASH_RESERVE_START_KB=16
env added CHIBIOS_BUILD_FLAGS=USE_FATFS=yes CHIBIOS_STARTUP_MK=os/common/startup/ARMCMx/compilers/GCC/mk/startup_stm32f4xx.mk CHIBIOS_PLATFORM_MK=os/hal/ports/STM32/STM32F4xx/platform.mk MCU=cortex-m4 ENV_UDEFS=-DCHPRINTF_USE_FLOAT=1
[3/8] Creating build/Pixhawk1/hwdef.h
[4/8] Creating build/Pixhawk1/modules/ChibiOS/include_dirs
[5/8] Processing modules/mavlink/message_definitions/v1.0/ardupilotmega.xml
Validation skipped for /tmp/ardupilot/modules/mavlink/message_definitions/v1.0/ardupilotmega.xml.
Parsing /tmp/ardupilot/modules/mavlink/message_definitions/v1.0/ardupilotmega.xml
Note: message DATA64 is longer than 64 bytes long (74 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message DATA96 is longer than 64 bytes long (106 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message REMOTE_LOG_DATA_BLOCK is longer than 64 bytes long (214 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message DEVICE_OP_READ_REPLY is longer than 64 bytes long (143 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message DEVICE_OP_WRITE is longer than 64 bytes long (187 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Validation skipped for /tmp/ardupilot/modules/mavlink/message_definitions/v1.0/common.xml.
Parsing /tmp/ardupilot/modules/mavlink/message_definitions/v1.0/common.xml
[6/8] Processing uavcangen: modules/uavcan/dsdl/uavcan/CoarseOrientation.uavcan modules/uavcan/dsdl/uavcan/Timestamp.uavcan modules/uavcan/dsdl/uavcan/equipment/actuator/1010.ArrayCommand.uavcan modules/uavcan/dsdl/uavcan/equipment/actuator/1011.Status.uavcan modules/uavcan/dsdl/uavcan/equipment/actuator/Command.uavcan modules/uavcan/dsdl/uavcan/equipment/ahrs/1000.Solution.uavcan modules/uavcan/dsdl/uavcan/equipment/ahrs/1001.MagneticFieldStrength.uavcan modules/uavcan/dsdl/uavcan/equipment/ahrs/1002.MagneticFieldStrength2.uavcan modules/uavcan/dsdl/uavcan/equipment/ahrs/1003.RawIMU.uavcan modules/uavcan/dsdl/uavcan/equipment/air_data/1020.TrueAirspeed.uavcan modules/uavcan/dsdl/uavcan/equipment/air_data/1021.IndicatedAirspeed.uavcan modules/uavcan/dsdl/uavcan/equipment/air_data/1025.AngleOfAttack.uavcan modules/uavcan/dsdl/uavcan/equipment/air_data/1026.Sideslip.uavcan modules/uavcan/dsdl/uavcan/equipment/air_data/1027.RawAirData.uavcan modules/uavcan/dsdl/uavcan/equipment/air_data/1028.StaticPressure.uavcan modules/uavcan/dsdl/uavcan/equipment/air_data/1029.StaticTemperature.uavcan modules/uavcan/dsdl/uavcan/equipment/camera_gimbal/1040.AngularCommand.uavcan modules/uavcan/dsdl/uavcan/equipment/camera_gimbal/1041.GEOPOICommand.uavcan modules/uavcan/dsdl/uavcan/equipment/camera_gimbal/1044.Status.uavcan modules/uavcan/dsdl/uavcan/equipment/camera_gimbal/Mode.uavcan modules/uavcan/dsdl/uavcan/equipment/device/1110.Temperature.uavcan modules/uavcan/dsdl/uavcan/equipment/esc/1030.RawCommand.uavcan modules/uavcan/dsdl/uavcan/equipment/esc/1031.RPMCommand.uavcan modules/uavcan/dsdl/uavcan/equipment/esc/1034.Status.uavcan modules/uavcan/dsdl/uavcan/equipment/gnss/1060.Fix.uavcan modules/uavcan/dsdl/uavcan/equipment/gnss/1061.Auxiliary.uavcan modules/uavcan/dsdl/uavcan/equipment/gnss/1062.RTCMStream.uavcan modules/uavcan/dsdl/uavcan/equipment/gnss/1063.Fix2.uavcan modules/uavcan/dsdl/uavcan/equipment/gnss/ECEFPositionVelocity.uavcan modules/uavcan/dsdl/uavcan/equipment/hardpoint/1070.Command.uavcan modules/uavcan/dsdl/uavcan/equipment/hardpoint/1071.Status.uavcan modules/uavcan/dsdl/uavcan/equipment/ice/1129.FuelTankStatus.uavcan modules/uavcan/dsdl/uavcan/equipment/ice/reciprocating/1120.Status.uavcan modules/uavcan/dsdl/uavcan/equipment/ice/reciprocating/CylinderStatus.uavcan modules/uavcan/dsdl/uavcan/equipment/indication/1080.BeepCommand.uavcan modules/uavcan/dsdl/uavcan/equipment/indication/1081.LightsCommand.uavcan modules/uavcan/dsdl/uavcan/equipment/indication/RGB565.uavcan modules/uavcan/dsdl/uavcan/equipment/indication/SingleLightCommand.uavcan modules/uavcan/dsdl/uavcan/equipment/power/1090.PrimaryPowerSupplyStatus.uavcan modules/uavcan/dsdl/uavcan/equipment/power/1091.CircuitStatus.uavcan modules/uavcan/dsdl/uavcan/equipment/power/1092.BatteryInfo.uavcan modules/uavcan/dsdl/uavcan/equipment/range_sensor/1050.Measurement.uavcan modules/uavcan/dsdl/uavcan/equipment/safety/1100.ArmingStatus.uavcan modules/uavcan/dsdl/uavcan/navigation/2000.GlobalNavigationSolution.uavcan modules/uavcan/dsdl/uavcan/protocol/1.GetNodeInfo.uavcan modules/uavcan/dsdl/uavcan/protocol/2.GetDataTypeInfo.uavcan modules/uavcan/dsdl/uavcan/protocol/341.NodeStatus.uavcan modules/uavcan/dsdl/uavcan/protocol/4.GetTransportStats.uavcan modules/uavcan/dsdl/uavcan/protocol/4.GlobalTimeSync.uavcan modules/uavcan/dsdl/uavcan/protocol/5.Panic.uavcan modules/uavcan/dsdl/uavcan/protocol/5.RestartNode.uavcan modules/uavcan/dsdl/uavcan/protocol/6.AccessCommandShell.uavcan modules/uavcan/dsdl/uavcan/protocol/CANIfaceStats.uavcan modules/uavcan/dsdl/uavcan/protocol/DataTypeKind.uavcan modules/uavcan/dsdl/uavcan/protocol/HardwareVersion.uavcan modules/uavcan/dsdl/uavcan/protocol/SoftwareVersion.uavcan modules/uavcan/dsdl/uavcan/protocol/debug/16370.KeyValue.uavcan modules/uavcan/dsdl/uavcan/protocol/debug/16383.LogMessage.uavcan modules/uavcan/dsdl/uavcan/protocol/debug/LogLevel.uavcan modules/uavcan/dsdl/uavcan/protocol/dynamic_node_id/1.Allocation.uavcan modules/uavcan/dsdl/uavcan/protocol/dynamic_node_id/server/30.AppendEntries.uavcan modules/uavcan/dsdl/uavcan/protocol/dynamic_node_id/server/31.RequestVote.uavcan modules/uavcan/dsdl/uavcan/protocol/dynamic_node_id/server/390.Discovery.uavcan modules/uavcan/dsdl/uavcan/protocol/dynamic_node_id/server/Entry.uavcan modules/uavcan/dsdl/uavcan/protocol/enumeration/15.Begin.uavcan modules/uavcan/dsdl/uavcan/protocol/enumeration/380.Indication.uavcan modules/uavcan/dsdl/uavcan/protocol/file/40.BeginFirmwareUpdate.uavcan modules/uavcan/dsdl/uavcan/protocol/file/45.GetInfo.uavcan modules/uavcan/dsdl/uavcan/protocol/file/46.GetDirectoryEntryInfo.uavcan modules/uavcan/dsdl/uavcan/protocol/file/47.Delete.uavcan modules/uavcan/dsdl/uavcan/protocol/file/48.Read.uavcan modules/uavcan/dsdl/uavcan/protocol/file/49.Write.uavcan modules/uavcan/dsdl/uavcan/protocol/file/EntryType.uavcan modules/uavcan/dsdl/uavcan/protocol/file/Error.uavcan modules/uavcan/dsdl/uavcan/protocol/file/Path.uavcan modules/uavcan/dsdl/uavcan/protocol/param/10.ExecuteOpcode.uavcan modules/uavcan/dsdl/uavcan/protocol/param/11.GetSet.uavcan modules/uavcan/dsdl/uavcan/protocol/param/Empty.uavcan modules/uavcan/dsdl/uavcan/protocol/param/NumericValue.uavcan modules/uavcan/dsdl/uavcan/protocol/param/Value.uavcan modules/uavcan/dsdl/uavcan/tunnel/2010.Broadcast.uavcan modules/uavcan/dsdl/uavcan/tunnel/63.Call.uavcan modules/uavcan/dsdl/uavcan/tunnel/Protocol.uavcan
Note: message GPS_STATUS is longer than 64 bytes long (109 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message ATTITUDE_QUATERNION_COV is longer than 64 bytes long (80 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
[7/8] Creating build/Pixhawk1/ap_version.h
Note: message GLOBAL_POSITION_INT_COV is longer than 64 bytes long (189 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message LOCAL_POSITION_NED_COV is longer than 64 bytes long (233 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message HIL_ACTUATOR_CONTROLS is longer than 64 bytes long (89 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message GLOBAL_VISION_POSITION_ESTIMATE is longer than 64 bytes long (124 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message VISION_POSITION_ESTIMATE is longer than 64 bytes long (124 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message VICON_POSITION_ESTIMATE is longer than 64 bytes long (124 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message HIGHRES_IMU is longer than 64 bytes long (70 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message HIL_SENSOR is longer than 64 bytes long (72 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message SIM_STATE is longer than 64 bytes long (92 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message FILE_TRANSFER_PROTOCOL is longer than 64 bytes long (262 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message HIL_STATE_QUATERNION is longer than 64 bytes long (72 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message LOG_DATA is longer than 64 bytes long (105 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message GPS_INJECT_DATA is longer than 64 bytes long (121 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message SERIAL_CONTROL is longer than 64 bytes long (87 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message ENCAPSULATED_DATA is longer than 64 bytes long (263 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message ATT_POS_MOCAP is longer than 64 bytes long (128 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message RESOURCE_REQUEST is longer than 64 bytes long (251 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message FOLLOW_TARGET is longer than 64 bytes long (101 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message CONTROL_SYSTEM_STATE is longer than 64 bytes long (108 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message AUTOPILOT_VERSION is longer than 64 bytes long (86 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message LANDING_TARGET is longer than 64 bytes long (68 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message GPS_INPUT is longer than 64 bytes long (71 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message GPS_RTCM_DATA is longer than 64 bytes long (190 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message HOME_POSITION is longer than 64 bytes long (68 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message SET_HOME_POSITION is longer than 64 bytes long (69 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message V2_EXTENSION is longer than 64 bytes long (262 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message PLAY_TUNE is longer than 64 bytes long (240 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message CAMERA_INFORMATION is longer than 64 bytes long (243 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message CAMERA_IMAGE_CAPTURED is longer than 64 bytes long (263 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message LOGGING_DATA is longer than 64 bytes long (263 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message LOGGING_DATA_ACKED is longer than 64 bytes long (263 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message WIFI_CONFIG_AP is longer than 64 bytes long (104 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message UAVCAN_NODE_INFO is longer than 64 bytes long (124 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message OBSTACLE_DISTANCE is longer than 64 bytes long (166 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message ODOMETRY is longer than 64 bytes long (238 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message DEBUG_FLOAT_ARRAY is longer than 64 bytes long (260 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message STATUSTEXT_LONG is longer than 64 bytes long (263 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Note: message WHEEL_DISTANCE is longer than 64 bytes long (145 bytes), which can cause fragmentation since many radio modems use 64 bytes as maximum air transfer unit.
Validation skipped for /tmp/ardupilot/modules/mavlink/message_definitions/v1.0/uAvionix.xml.
Parsing /tmp/ardupilot/modules/mavlink/message_definitions/v1.0/uAvionix.xml
Validation skipped for /tmp/ardupilot/modules/mavlink/message_definitions/v1.0/icarous.xml.
Parsing /tmp/ardupilot/modules/mavlink/message_definitions/v1.0/icarous.xml
Merged enum MAV_CMD
Found 222 MAVLink message types in 4 XML files
Generating C implementation in directory /tmp/ardupilot/build/Pixhawk1/libraries/GCS_MAVLink/include/mavlink/v2.0/ardupilotmega
Including /tmp/ardupilot/libraries/AP_HAL_ChibiOS/hwdef/fmuv3/hwdef.dat
Setup for MCU STM32F427xx
Writing hwdef setup in /tmp/ardupilot/build/Pixhawk1/hwdef.h
Writing DMA map
Generating ldscript.ld

Generating C implementation in directory /tmp/ardupilot/build/Pixhawk1/libraries/GCS_MAVLink/include/mavlink/v2.0/common
Generating C implementation in directory /tmp/ardupilot/build/Pixhawk1/libraries/GCS_MAVLink/include/mavlink/v2.0/uAvionix
Generating C implementation in directory /tmp/ardupilot/build/Pixhawk1/libraries/GCS_MAVLink/include/mavlink/v2.0/icarous
Copying fixed headers for protocol 2.0 to /tmp/ardupilot/build/Pixhawk1/libraries/GCS_MAVLink/include/mavlink/v2.0
Compiler failure
Traceback (most recent call last):
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 300, in enum_last_value
    last = next(it)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 107, in run_generator
    text = generate_one_type(template_expander, t)
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 239, in generate_one_type
    text = template_expander(t=t)  # t for Type
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 307, in expand
    return template(**args)
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1030, in __call__
    u = "".join(self._render(self.parsetree, self.current_data))
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1130, in render
    output.extend(self.render(elem[2], data))
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1104, in render
    output.append(unicode(_eval(elem[1], data)))
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1079, in _eval
    return self.evalfunc(expr, data)
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 884, in eval
    x = eval(self.compile(expr), {"__builtins__":self.eval_allowed_globals}, locals)
  File "", line 1, in <module>
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1030, in __call__
    u = "".join(self._render(self.parsetree, self.current_data))
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1104, in render
    output.append(unicode(_eval(elem[1], data)))
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1079, in _eval
    return self.evalfunc(expr, data)
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 884, in eval
    x = eval(self.compile(expr), {"__builtins__":self.eval_allowed_globals}, locals)
  File "", line 1, in <module>
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1030, in __call__
    u = "".join(self._render(self.parsetree, self.current_data))
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1138, in render
    output.extend(self.render(elem[1], data))
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1119, in render
    for i in loop_iter:
RuntimeError: generator raised StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdlc", line 61, in <module>
    dsdlc_run(args.source_dir, args.incdir, args.outdir)
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 62, in run
    run_generator(types, output_dir)
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 111, in run_generator
    die(ex)
  File "/tmp/ardupilot/modules/uavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 89, in die
    raise DsdlCompilerException(str(text))
libuavcan_dsdl_compiler.DsdlCompilerException: generator raised StopIteration
generator raised StopIteration

uavcangen returned 1 error code
Waf: Leaving directory `/tmp/ardupilot/build/Pixhawk1'
Build failed
 -> task in 'uavcan' failed (exit status 1): 
        {task 139758841556328: uavcangen CoarseOrientation.uavcan,Timestamp.uavcan,1010.ArrayCommand.uavcan,1011.Status.uavcan,Command.uavcan,1000.Solution.uavcan,1001.MagneticFieldStrength.uavcan,1002.MagneticFieldStrength2.uavcan,1003.RawIMU.uavcan,1020.TrueAirspeed.uavcan,1021.IndicatedAirspeed.uavcan,1025.AngleOfAttack.uavcan,1026.Sideslip.uavcan,1027.RawAirData.uavcan,1028.StaticPressure.uavcan,1029.StaticTemperature.uavcan,1040.AngularCommand.uavcan,1041.GEOPOICommand.uavcan,1044.Status.uavcan,Mode.uavcan,1110.Temperature.uavcan,1030.RawCommand.uavcan,1031.RPMCommand.uavcan,1034.Status.uavcan,1060.Fix.uavcan,1061.Auxiliary.uavcan,1062.RTCMStream.uavcan,1063.Fix2.uavcan,ECEFPositionVelocity.uavcan,1070.Command.uavcan,1071.Status.uavcan,1129.FuelTankStatus.uavcan,1120.Status.uavcan,CylinderStatus.uavcan,1080.BeepCommand.uavcan,1081.LightsCommand.uavcan,RGB565.uavcan,SingleLightCommand.uavcan,1090.PrimaryPowerSupplyStatus.uavcan,1091.CircuitStatus.uavcan,1092.BatteryInfo.uavcan,1050.Measurement.uavcan,1100.ArmingStatus.uavcan,2000.GlobalNavigationSolution.uavcan,1.GetNodeInfo.uavcan,2.GetDataTypeInfo.uavcan,341.NodeStatus.uavcan,4.GetTransportStats.uavcan,4.GlobalTimeSync.uavcan,5.Panic.uavcan,5.RestartNode.uavcan,6.AccessCommandShell.uavcan,CANIfaceStats.uavcan,DataTypeKind.uavcan,HardwareVersion.uavcan,SoftwareVersion.uavcan,16370.KeyValue.uavcan,16383.LogMessage.uavcan,LogLevel.uavcan,1.Allocation.uavcan,30.AppendEntries.uavcan,31.RequestVote.uavcan,390.Discovery.uavcan,Entry.uavcan,15.Begin.uavcan,380.Indication.uavcan,40.BeginFirmwareUpdate.uavcan,45.GetInfo.uavcan,46.GetDirectoryEntryInfo.uavcan,47.Delete.uavcan,48.Read.uavcan,49.Write.uavcan,EntryType.uavcan,Error.uavcan,Path.uavcan,10.ExecuteOpcode.uavcan,11.GetSet.uavcan,Empty.uavcan,NumericValue.uavcan,Value.uavcan,2010.Broadcast.uavcan,63.Call.uavcan,Protocol.uavcan -> }
 (run with -v to display more information)
```

**Comment by patrickelectric** [comment]
running waf with python2 fixed the problem.

**Comment by WickedShell** [comment]
@patrickelectric yeah this is due to [PEP 479](https://www.python.org/dev/peps/pep-0479/) I'd love to see this solved as it's forcing me down the Python2 road as well that you found.

**Comment by DaHaiHuha** [comment]
yes, i have come cross the same question, but how to use Python2 to build?

**Comment by patrickelectric** [comment]
`./ waf ... -python=python2`

**Comment by OXINARF** [comment]
Fixed with #11144.
